CDCgov/prime-reportstream
# feat: implement save row functionality in Table component [experience-5442]

This PR completes building out functionality within the Table component that allows for updated data for a row to be saved / persisted.

Note that the first commit contains all of the functional changes, the other two commits contain tests or test related refactors.

## Test Steps:

1. start a local frontend server: `yarn run start:localdev` from `frontend-react` directory
2. start a local backend server: `./gradlew run` from `prime-router` directory
3. visit http://localhost:3000/admin/value-sets/name, logging in as necessary
4. click `Edit` on one of the rows in the table
5. open your browser's dev tools console
6. edit a value in the editable row
7. click save
8. _VERIFY_: you see a log in your console with an object representing the data for the row you edited, including the value that you edited. See screenshot
9. refresh the page
10. click an `edit` button to enter edit mode
11. click save without editing
8. _VERIFY_: you see a log in your console saying `no data to save`

## Changes
* in `ColumnData`, let parent component know about changes to field values during editing
* in `TableRows`, run the passed in callback function to allow for persisting of changes
* in value sets components, pass a function to table that will log out potential changes for edits (actual API interaction to come later)
* refactors to allow for tests of ColumnData and TableRows changes - split these component out of the Table component render
* tests

## Checklist

### Testing

### Process

## Linked Issues
- contributes to https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/cdcgov/prime-reportstream/5442

## To Be Done
- see `Work Breakdown` https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/cdcgov/prime-reportstream/5442
